proxy/bridge/terminal.go

```diff
@@ -2,208 +2,137 @@ package bridge
+)
 
 type TerminalBridge struct {
 	TlsConfig *tls.Config
 	Log       *logrus.Logger
 }
 
-type GRPCEndpoint struct {
-	SessionID string `json:"sessionID"`
-}
+func copyMsgs(writeMutex *sync.Mutex, dest, src *websocket.Conn) error {
+	for {
+		messageType, msg, err := src.ReadMessage()
+		if err != nil {
+			return err
+		}
 
-func (t TerminalBridge) HandleTerminal(w http.ResponseWriter, r *http.Request) {
-	deviceId, found := strings.CutPrefix(r.URL.Path, "/api/terminal/")
-	if !found {
-		t.Log.Warnf("Failed to get deviceId: %s", deviceId)
-		w.WriteHeader(http.StatusNotFound)
-		return
+		if writeMutex == nil {
+			err = dest.WriteMessage(messageType, msg)
+		} else {
+			writeMutex.Lock()
+			err = dest.WriteMessage(messageType, msg)
+			writeMutex.Unlock()
+		}
+
+		if err != nil {
+			return err
+		}
 	}
+}
 
-	client := &http.Client{Transport: &http.Transport{
-		TLSClientConfig: t.TlsConfig,
-	}}
+func (t TerminalBridge) HandleTerminal(w http.ResponseWriter, r *http.Request) {
+	isWebsocket := false
+	upgrades := r.Header["Upgrade"]
 
-	consoleUrl := config.FctlApiUrl + "/api/v1/devices/" + deviceId + "/console"
+	for _, upgrade := range upgrades {
+		if strings.ToLower(upgrade) == "websocket" {
+			isWebsocket = true
+			break
+		}
+	}
 
-	req, err := http.NewRequest(http.MethodGet, consoleUrl, nil)
-	if err != nil {
-		errMsg := fmt.Sprintf("Could not create request: %s", err.Error())
+	if !isWebsocket {
+		errMsg := "not a websocket connection"
 		t.Log.Warnf(errMsg)
 		w.WriteHeader(http.StatusInternalServerError)
 		w.Write([]byte(errMsg))
-		return
 	}
 
-	token := ""
-	authHeader, ok := r.Header[common.AuthHeaderKey]
-	if ok && len(authHeader) == 1 {
-		token = authHeader[0]
-		req.Header.Set(common.AuthHeaderKey, authHeader[0])
-	}
-
-	resp, err := client.Do(req)
-	if err != nil {
-		errMsg := fmt.Sprintf("Failed to get terminal session for device %s", deviceId)
-		t.Log.Warnf(errMsg)
+	deviceId, found := strings.CutPrefix(r.URL.Path, "/api/terminal/")
+	if !found {
+		t.Log.Warnf("Failed to get deviceId: %s", deviceId)
 		w.WriteHeader(http.StatusNotFound)
-		w.Write([]byte(errMsg))
-		return
-	}
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		errMsg := fmt.Sprintf("Failed to read terminal session response %s", err.Error())
-		t.Log.Warnf(errMsg)
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte(errMsg))
-		return
-	}
-	if resp.StatusCode != http.StatusOK {
-		t.Log.Warnf("Unexpected terminal session response code %d", resp.StatusCode)
-		w.WriteHeader(resp.StatusCode)
-		w.Write(body)
 		return
 	}
-	response := GRPCEndpoint{}
-	if err := json.Unmarshal(body, &response); err != nil {
-		errMsg := fmt.Sprintf("Failed to unmarshall response json: %s", err.Error())
-		t.Log.Warnf(errMsg)
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte(errMsg))
-		return
+
+	consoleUrl := fmt.Sprintf("%s/api/v1/ws/devices/%s/console", config.FctlApiUrl, deviceId)
```

Comment:
```suggestion
	consoleUrl := fmt.Sprintf("%s/ws/v1/devices/%s/console", config.FctlApiUrl, deviceId)
```

Sorry, this was my fault :) I communicate it incorrectly

Reply:
ah, thanks!